pkg/build/builder/cmd/scmauth/util.go

```diff
@@ -0,0 +1,68 @@
+package scmauth
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func createGitConfig(includePath string) error {
+	tempDir, err := ioutil.TempDir("", "git")
+	if err != nil {
+		return err
+	}
+	gitconfig := filepath.Join(tempDir, ".gitconfig")
+	content := fmt.Sprintf("[include]\npath = %s\n", includePath)
+	if err := ioutil.WriteFile(gitconfig, []byte(content), 0600); err != nil {
+		return err
+	}
+	// The GIT_CONFIG variable won't affect regular git operation
+	// therefore the HOME variable needs to be set so git can pick up
+	// .gitconfig from that location. The GIT_CONFIG variable is still used
+	// to track the location of the GIT_CONFIG for multiple SCMAuth objects.
+	os.Setenv("HOME", tempDir)
+	os.Setenv("GIT_CONFIG", gitconfig)
+	return nil
+}
+
+func readFile(fileName string) ([]string, error) {
+	file, err := os.Open(fileName)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	var lines []string
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+	return lines, scanner.Err()
+}
+
+// ensureGitConfig ensures that the OS env var GIT_CONFIG is set and
+// that it points to a file that has an include statement for the given path
+func ensureGitConfig(includePath string) error {
+	gitconfig := os.Getenv("GIT_CONFIG")
+	if gitconfig == "" {
+		return createGitConfig(includePath)
+	}
+
+	lines, err := readFile(gitconfig)
+	if err != nil {
+		return err
+	}
+	for _, line := range lines {
+		// If include already exists, return with no error
+		if line == fmt.Sprintf("path = %s", includePath) {
+			return nil
+		}
+	}
+
+	lines = append(lines, fmt.Sprintf("path = %s", includePath))
```

Comment:
i guess maybe not the way you're using it..I was thinking you might be calling this with an existing gitconfig file that might or might not end in a [include] stanza but it looks like you only use it when it's a gitconfig created by the createGitConfig method.